Classes/ScrnBalance.uc

```diff
@@ -3205,51 +3034,83 @@ function ServerTraveling(string URL, bool bItems)
     if (NextMutator != None)
         NextMutator.ServerTraveling(URL,bItems);
 
-    if ( ScrnGT == none || ScrnGT.ScrnGameLength == none )
+    Cleanup();
+}
+
+simulated function Destroyed()
+{
+    log("ScrnBalance destroyed", 'ScrnBalance');
+    super.Destroyed();
+}
+
+
+final protected simulated function Cleanup()
```

Comment:
No need to introduce `Cleanup()` function because the code cannot be executed anywhere but inside `ServerTraveling()`. Please restore the original `ServerTraveling` code. No changes are needed there. Let's follow the _"if it ain't broke, don't fix it"_ rule.